ex_webrtc/lib/ex_webrtc/peer_connection_handler.ex

```diff
@@ -290,8 +321,14 @@ defmodule Membrane.RTC.Engine.Endpoint.ExWebRTC.PeerConnectionHandler do
     {actions, state}
   end
 
-  defp handle_webrtc_msg({:signaling_state_change, :stable}, _ctx, state) do
-    {[notify_parent: :negotiation_done], state}
+  defp handle_webrtc_msg({:signaling_state_change, new_state}, _ctx, state) do
+    actions =
+      case {state.peer_connection_signaling_state, new_state} do
+        {:have_remote_offer, :stable} -> [notify_parent: :negotiation_done]
+        _other -> []
+      end
+
+    {actions, %{state | peer_connection_signaling_state: new_state}}
```

Comment:
Could you tell me something more about this part?

Reply:
The change of state to: `{:signaling_state_change, :stable}` means, that the connection is in a stable state (i.e. there are no ongoing negotiations, tracks added/ removed etc.)

This usually means, that the ongoing negotiation has been completed.
However, there is a single instance when this event is received, even though there has been no negotiation going on. That happens at the very beginning. Upon creation the PeerConnection has state `:new`, which then changes to `:stable` (even if no tracks are negotiated). So far we ignored this case, which was a bug. Now it's fixed, by checking if there has been a negotiation going on before receiving the `:stable` state. 

For more info check MDN docs: https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/signalingState